src/CommandRunner.cpp

```diff
@@ -311,35 +311,41 @@ CommandRunner::CommandRunner(bool debug,
  *
  * @return The output of the command read from the output file.
  */
-Try<std::string> CommandRunner::run(const Command& command,
-                                    const std::string& input) {
+Try<string> CommandRunner::run(const Command& command,
+                               const std::string& input) {
   try {
-    TemporaryFile inputFile;
-    TemporaryFile outputFile;
-    inputFile.write(input);
+    TemporaryFile stdinFile;
+    TemporaryFile stdoutFile;
+    TemporaryFile stderrFile;
+    stdinFile.write(input);
 
     if (m_debug) {
       TASK_LOG(INFO, m_loggingMetadata)
           << "Calling command: \"" << command.command() << "\" ("
-          << command.timeout() << "s) " << inputFile.filepath() << " "
-          << outputFile.filepath();
+          << command.timeout() << "s) " << stdinFile.filepath() << " "
+          << stdoutFile.filepath() << " " << stderrFile.filepath();
     } else {
       TASK_LOG(INFO, m_loggingMetadata)
           << "Calling command: \"" << command.command() << "\" ("
           << command.timeout() << "s)";
     }
 
     vector<string> args;
-    args.push_back(inputFile.filepath());
-    args.push_back(outputFile.filepath());
+    args.push_back(stdinFile.filepath());
+    args.push_back(stdoutFile.filepath());
+    args.push_back(stderrFile.filepath());
 
     Try<Nothing> result = runCommandWithTimeout(
         command.command(), args, command.timeout(), m_loggingMetadata);
     if (result.isError()) {
+      string stderr = stderrFile.readAll();
+      if (!stderr.empty()) {
+        throw std::runtime_error(result.error() + " Cause: " + stderr);
+      }
       throw std::runtime_error(result.error());
     }
 
-    return outputFile.readAll();
+    return stdoutFile.readAll();
```

Comment:
maybe we should return a concatenation of stdout and stderr. Otherwise mesos logs will only show "normal" output and not teh one that is interesting for debug

Reply:
As mentioned in previous comment stdout and stderr are badly named since they are not log containers but output buffers (containing the returned JSON and the exception message). I will rename them.

Reply:
fixed